WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.kt

```diff
@@ -0,0 +1,130 @@
+import org.wordpress.android.fluxc.model.TermModel
+import org.wordpress.android.fluxc.store.TaxonomyStore
+import org.wordpress.android.models.CategoryNode
+import org.wordpress.android.util.ToastUtils
+import javax.inject.Inject
+
+class AddCategoryFragment : AppCompatDialogFragment() {
+    private var site: SiteModel? = null
+    private var binding: AddCategoryBinding? = null
+
+    @set:Inject
+    var mTaxonomyStore: TaxonomyStore? = null
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        (requireActivity().application as WordPress).component().inject(this)
+        initSite(savedInstanceState)
+        val builder =
+            MaterialAlertDialogBuilder(ContextThemeWrapper(activity, R.style.PostSettingsTheme))
+        binding = AddCategoryBinding.inflate(layoutInflater, null, false)
+        loadCategories()
+        builder.setView(binding!!.root)
+            .setPositiveButton(android.R.string.ok, null)
+            .setNegativeButton(android.R.string.cancel, null)
+
+        return builder.create()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val dialog = requireDialog() as AlertDialog
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
+            if (addCategory()) {
+                dismiss()
+            }
+        }
+    }
+
+    @Suppress("Deprecation")
+    private fun initSite(savedInstanceState: Bundle?) {
+        site = if (savedInstanceState == null) {
+            if (arguments != null) {
+                requireArguments().getSerializable(WordPress.SITE) as SiteModel?
+            } else {
+                requireActivity().intent.getSerializableExtra(WordPress.SITE) as SiteModel?
+            }
+        } else {
+            savedInstanceState.getSerializable(WordPress.SITE) as SiteModel?
+        }
+
+        if (site == null) {
+            ToastUtils.showToast(
+                requireActivity(),
+                R.string.blog_not_found,
+                ToastUtils.Duration.SHORT
+            )
+            parentFragmentManager.popBackStack()
+        }
+    }
+
+    private fun addCategory(): Boolean {
+        val categoryName = binding?.categoryName?.text.toString()
+        val selectedCategory = binding?.parentCategory?.selectedItem as CategoryNode
+        val parentId = selectedCategory.categoryId
+
+        if (categoryName.replace(" ".toRegex(), "") == "") {
+            binding!!.categoryName.error = getText(R.string.cat_name_required)
+            return false
+        }
+
+        val newCategory = TermModel(
+            TaxonomyStore.DEFAULT_TAXONOMY_CATEGORY,
+            categoryName,
+            parentId
+        )
+        (requireActivity() as SelectCategoriesActivity).categoryAdded(newCategory)
+        return true
+    }
+
+    private fun loadCategories() {
+        val rootCategory = CategoryNode.createCategoryTreeFromList(
+            mTaxonomyStore!!.getCategoriesForSite(
+                site!!
```

Comment:
We should avoid using `!!`.  In `onCreate` we do this
```
if (site == null) {
            ToastUtils.showToast(
                requireActivity(),
                R.string.blog_not_found,
                ToastUtils.Duration.SHORT
            )
            parentFragmentManager.popBackStack()
        }
```

So, we can do the same here if site or `mTaxonomyStore` is null.

Reply:
My apologies! I quite didn't understood this?

Reply:
Let me explain it more clearly. If either site or mTaxonomyStore is null, calling mTaxonomyStore!!.getCategoriesForSite(site!!) will cause a crash. I suggest the following changes:
- Remove the !! operators from the loadCategories function.
- Declare mTaxonomyStore as lateinit because we use store classes like this throughout the codebase.

See this patch:

[Remove_!!_from_loadCategories()_function.patch](https://github.com/user-attachments/files/15888452/Remove_._from_loadCategories._function.patch)